app/org/maproulette/models/Task.scala

```diff
@@ -256,26 +256,40 @@ object Task extends CommonField {
     * If current status is skipped, then it can set the status to fixed, false_positive or deleted
     * If current statis is deleted, then it can set the status to created. Essentially resetting the task
     *
+    * The exception is if we are allowing the task status to be changed by the mapper in case
+    * of review revisions and then these completed statuses can also change to another completed status:
+    * fixed, false_positive, too_hard, already_fixed
+    *
     * @param current The current status of the task
     * @param toSet   The status that the task will be set too
+    * @param allowChange Allow an already completed status to be changed to another completed status.
     * @return True if the status can be set without violating any of the above rules
     */
-  def isValidStatusProgression(current: Int, toSet: Int): Boolean = {
+  def isValidStatusProgression(current: Int, toSet: Int, allowChange: Boolean = false): Boolean = {
```

Comment:
I don't notice any test updates for this, and if we don't have any tests we definitely should.

Reply:
Added TaskSpec and tests for this method.